src/cmd/build.js

```diff
@@ -37,14 +39,55 @@ export type PackageCreatorParams = {
   artifactsDir: string,
 };
 
+export type LocalizedNameParams = {
+  messageFile: string,
+  manifestData: ExtensionManifest,
+}
+
 export type PackageCreatorFn =
     (params: PackageCreatorParams) => Promise<ExtensionBuildResult>;
 
+export async function getDefaultLocalizedName(
+  {messageFile, manifestData}: LocalizedNameParams
+): Promise<string> {
+
+  let messageData: string|Buffer;
+  let extensionName: string = manifestData.name;
+
+  try {
+    messageData = await fs.readFile(messageFile);
+  }
+  catch (err) {
+    log.info(err);
+    throw new WebExtError(`${err}`);
+  }
+
+  try {
+    messageData = JSON.parse(messageData);
+  }
+  catch (err) {
+    throw new WebExtError(`The JSON file is malformed: ${err}`);
+  }
+  extensionName = manifestData.name.replace(/__MSG_([A-Za-z0-9@_]+?)__/g,
+    (match, messageName) => {
+      if (!(messageData[messageName]
+            && messageData[messageName].message)) {
+        const error = new WebExtError
+          ('The locale file messages.json ' +
+            `for key: ${messageName}`);
```

Comment:
I don't understand this message. I think it should be something more like this, right? `The locale file ${messageFile} is missing key ${messageName}`

Also, it can be a `UsageError` since this is fixable by the developer. This error means they tried to use a localization key that hasn't been defined.